specs/models/DataHubEnvelope.schema.json

```diff
@@ -0,0 +1,141 @@
+  "type": "object",
+  "properties": {
+    "headers": {
+      "description": "The unique path to this property. For a structured property, uses dot notation - e.g. billing.street.fiveDigit",
+      "type": "object",
+      "properties": {
+        "createdBy": {
+          "description": "MarkLogic user that created this document",
+          "type": "string"
+        },
+        "createdOn": {
+          "description": "dateTime at which this document was created",
+          "type": "string"
+        },
+        "createdUsingFile": {
+          "description": "Will be set when running an ingestion step via DHF and ingesting CSV files",
+          "type": "string"
+        },
+        "datahub": {
+          "description": "Intended to store all DHF-specific headers in the future",
+          "type": "object",
+          "properties": {
+            "validationErrors": {
+              "description": "Populated by a mapping step that has entity validation enabled",
+              "properties": {
+                "name": {
+                  "type": "string"
+                },
+                "data": {
+                  "type": "object"
+                },
+                "message": {
+                  "type": "string"
+                },
+                "formattedMessages": {
+                  "type": "array",
+                  "items": {
+                    "type": "string"
+                  }
+                }
+              }
+            }
+          }
+        },
+        "id": {
+          "description": "Appears to be set as part of mastering",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "merges": {
+          "type": "array",
+          "items": {
+            "type": "object",
+            "properties": {
+              "document-uri": {
+                "type": "string"
+              },
+              "last-merge": {
+                "type": "string"
+              }
+            }
+          }
+        },
+        "merge-options": {
+          "type": "object",
+          "properties": {
+            "lang": {
+              "type": "string"
+            },
+            "value": {
+              "type": "string"
+            }
+          }
+        },
+        "sources": {
+          "type": "array",
+          "items": {
+            "type": "object",
+            "properties": {
+              "dateTime": {
+                "description": "May be specific to mastering",
+                "type": "string"
+              },
+              "import-id": {
+                "description": "Appears to be specific to mastering",
+                "type": "string"
+              },
+              "name": {
+                "description": "Name of a source that provides instance data in this envelope",
+                "type": "string"
+              },
+              "user": {
+                "description": "May be specific to mastering",
```

Comment:
This isn't used by mastering for anything. Other properties can exist in the source object, but name and dateTime are the only properties that have features with some dependency on them.